frappe/frappe
# fix: user permission for restricting nested struct

Adding hide descendants to the or_filters is pulling in all records that have hide descendants turned even though they might not be on the same doctype
Hide descendants should not be considered while adding user permissions since a permission with and one without it is not useful.

How to test?
Create user permission with hide Descendents checked and apply to all doctypes not checked
Now create another user permission with hide Descendents checked but apply it to a different doctype
This should not raise a validation about having two default user permissions
